crates/red_knot_python_semantic/src/types/class.rs

```diff
@@ -1557,4 +1656,35 @@ mod tests {
             );
         }
     }
+
+    fn setup_db_with_broken_typeshed(builtins_file: &str) -> TestDb {
```

Comment:
I guess this is a wording nit, sort of related to my "not sure we need to warn" comment above -- but I slightly object to the framing of this as a "broken" typeshed. I think we should aim to consider typesheds where certain symbols aren't present or aren't what we expect to still be "working", where some behaviors will just fall back to gradual typing. Maybe it doesn't really matter, but I'd prefer we don't accept the idea that it's OK for things to be "broken" because a minimal typeshed is used.

Reply:
Fair enough. FWIW, we _do_ panic (Salsa cycles etc.) if there's no `object` class in builtins; we did before this PR, and will continue to with this PR. I'm not sure I have the appetite to fix that "bug" :P

Reply:
If it's salsa cycles, I expect the panics will go away with fixpoint iteration? Will check! Not having `object` basically means everything is `Unknown` (because everything inherits `Unknown`)...